src/modules/account/RadixAccount.ts

```diff
@@ -183,15 +215,96 @@ export default class RadixAccount {
     }
 
     private _onAtomReceived = async (atomObservation: RadixAtomObservation) => {
-        this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
+
+
+        if (!this.processingAtomCounter.closed && !this.processingAtomCounter.isStopped) {
+            this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
+        } else {
+            logger.error(`☢️ processingAtomCounter closed or stopped`)
+        }
 
         atomObservation.processedData = {}
         for (const system of this.accountSystems.values()) {
             await system.processAtomUpdate(atomObservation)
         }
 
-        this.processingAtomCounter.next(this.processingAtomCounter.getValue() - 1)
+        if (!this.processingAtomCounter.closed && !this.processingAtomCounter.isStopped) {
+            this.processingAtomCounter.next(this.processingAtomCounter.getValue() - 1)
+        } else {
+            logger.error(`☢️ processingAtomCounter closed or stopped`)
+        }
+
+    }
+
+    public async requestRadsForDevelopmentFromFaucetService(): Promise<RadixTransaction> {
+        return this.requestRadsForDevelopmentFromFaucetServiceWithoutBalanceUpdate()
+            .then((txUnique) => {
+                    return this.transferSystem.getAllTransactions()
+                        .delay(2_000)
+                        .take(1)
+                        .timeout(60_000)
+                        .pipe(catchError((error, obs) => {
+                            throw new Error(`🚰 Failed to update balance after having received test XRD tokens from
+                              faucet in tx with unique string
+                               ${txUnique}, error: ${error}
+                               `)
+                        }))
+                        .pipe(map((tu: RadixTransactionUpdate) => tu.transaction))
+                        .toPromise()
+                },
+            ).catch(errorFromFaucet => {
+                const errorMsg = `Faucet server is down? ${errorFromFaucet}`
+                logger.error(errorMsg)
+                throw new Error(errorMsg)
+            })
+    }
+
+    public async requestRadsForDevelopmentFromFaucetServiceWithoutBalanceUpdate(
+        maxNumberOfRetries: number = 10,
+        killNodeServerOfFailure: boolean = true,
+    ): Promise<string> {
+
+        for (let attempt = 0; attempt < maxNumberOfRetries; attempt++) {
+            const idOfTxFromFaucet = await RadixAccount.getTokensFromFaucetURL(this.address)
+            if (idOfTxFromFaucet !== undefined) {
+                return idOfTxFromFaucet
+            }
+            logger.error(`Sleeping for 1second after having failed to get tokens from faucet`)
+            await sleep(1000)
```

Comment:
Would suggest that we use a linear backoff here, as a constant backoff may be sufficient to keep the faucet in a bad state with balance updates.

Reply:
yupp, thought of that earlier, lets do it